google/cloud/dataproc_spark_connect/session.py

```diff
@@ -407,9 +409,29 @@ def _get_dataproc_config(self):
                     )
                 }
             if "COLAB_NOTEBOOK_ID" in os.environ:
-                dataproc_config.labels["colab-notebook-id"] = os.environ[
-                    "COLAB_NOTEBOOK_ID"
-                ]
+                colab_notebook_id_full = os.environ["COLAB_NOTEBOOK_ID"]
+                # Expected format: /embedded/projects/proj-id/locations/us-central1/repositories/notebook-id
+                match = re.search(
+                    r"/projects/(?P<project>[^/]+)/locations/(?P<location>[^/]+)/repositories/(?P<id>[^/]+)",
+                    colab_notebook_id_full,
+                )
+                if match:
+                    gd = match.groupdict()
+                    if "project" in gd and "location" in gd and "id" in gd:
```

Comment:
I feel that it became increasingly hacky - we may want to consider more systematic approach and and new "owner" API field for this that can take whole notebok name.